certora/specs/Splits.spec

```diff
@@ -0,0 +1,682 @@
+    uint128 userB_splittableAfter = splittable(e, userB_Id, assetId);
+    uint128 userB_collectableAfter = collectable(e, userB_Id, assetId);
+    uint128 userC_splittableAfter = splittable(e, userC_Id, assetId);
+    uint128 userC_collectableAfter = collectable(e, userC_Id, assetId);
+
+
+    // the expectation:
+    // UserA's splittable should NOT increase
+    assert userA_splittableAfter <= userA_splittableBefore;
+
+    // UserB is on the list of UserA's splitters, therefore "userB's splittable should NOT decrease"
+    assert userB_splittableAfter >= userB_splittableBefore;
+
+    // UserC is NOT on the list of UserA's splitters, therefore "userC's splittable should NOT change"
+    assert userC_splittableAfter == userC_splittableBefore;
+
+    // UserA's collectable should NOT decrease, UserB and UserC's collectable should NOT change
+    assert userA_collectableAfter >= userA_collectableBefore;
+    assert userB_collectableAfter == userB_collectableBefore;
+    assert userC_collectableAfter == userC_collectableBefore;
+}
+
+
+/// @notice Different assets should not interfere with each other
+/// operations over one assetId1 should not affect anything related to other assetId2
+/// calling split on userA with assetId1 should NOT affect 
+/// the splittable and collectable for any user's assetId2
+rule assetsDoNotInterfereEachOther() {
+    env e; uint256 assetId1; uint256 assetId2; uint256 userA_Id; uint256 userB_Id;
+
+    // make sure the users/assets are not the same
+    require userA_Id != userB_Id;
+    require assetId1 != assetId2;
+
+    // recording the state before split()
+    uint128 userA_splittableAssetId1_Before = splittable(e, userA_Id, assetId1);
+    uint128 userA_collectableAssetId1_Before = collectable(e, userA_Id, assetId1);
+    uint128 userA_splittableAssetId2_Before = splittable(e, userA_Id, assetId2);
+    uint128 userA_collectableAssetId2_Before = collectable(e, userA_Id, assetId2);
+
+    uint128 userB_splittableAssetId1_Before = splittable(e, userB_Id, assetId1);
+    uint128 userB_collectableAssetId1_Before = collectable(e, userB_Id, assetId1);
+    uint128 userB_splittableAssetId2_Before = splittable(e, userB_Id, assetId2);
+    uint128 userB_collectableAssetId2_Before = collectable(e, userB_Id, assetId2);
+
+    // calling the split() for userA over assertId1
+    uint128 userA_collectableAmtAssetId1; uint128 userA_splitAmtAssetId1;
+    userA_collectableAmtAssetId1, userA_splitAmtAssetId1 = split(e, userA_Id, assetId1, true);
+
+    // recording the state after split()
+    uint128 userA_splittableAssetId1_After = splittable(e, userA_Id, assetId1);
+    uint128 userA_collectableAssetId1_After = collectable(e, userA_Id, assetId1);
+    uint128 userA_splittableAssetId2_After = splittable(e, userA_Id, assetId2);
+    uint128 userA_collectableAssetId2_After = collectable(e, userA_Id, assetId2);
+
+    uint128 userB_splittableAssetId1_After = splittable(e, userB_Id, assetId1);
```

Comment:
Note that you're not using this value and the one below it.
It's advised to check that for the same assetId1, the balance of userB_ID isn't changed as well.
You need to make sure, of course, that userB_ID is not any of the parties involved.